src/components/smart_objects/include/smart_objects/schema_item.h

```diff
@@ -91,13 +101,27 @@ class ISchemaItem {
    **/
   virtual bool hasDefaultValue(SmartObject& Object);
 
+  /**
+   * @brief Filter invalid enum values
+   *
+   * @param Object Object to check for invalid enum values
+   * @param MessageVersion the version of the schema to use for validation
+   * @param report__ object for reporting enums which were removed during the
+   * process
+   *
+   * @return true if the value being checked should be filtered, false otherwise
```

Comment:
The return value is a bit confusing. It says if true it should be filtered, but isnt the enum implementation of this method doing the filtering?